nncf/tensorflow/quantization/algorithm.py

```diff
@@ -368,71 +357,152 @@ def _run_batchnorm_adaptation(self, model: tf.keras.Model) -> None:
+                    target_point = TFLayerWeight(layer_info.layer_name, weight_def.weight_attr_name)
+                    qpoint = TFQuantizationPoint(op_name, quantizer_spec, target_point)
+                    setup.add_quantization_point(qpoint)
+            else:
+                assert qp.is_activation_quantization_point()
+                ip = qp.insertion_point
+                assert isinstance(ip, ActivationQuantizationInsertionPoint)
+                target_node_name = ip.target_node_name
+                input_port_id = ip.input_port_id
+                fake_quantize_name = self._get_fake_quantize_name(target_node_name, input_port_id)
+                quantizer_spec = TFQuantizerSpec.from_config(qp.qconfig, narrow_range=False, half_range=False)
+                fake_quantize_layer = FakeQuantize(
+                    quantizer_spec,
+                    name=fake_quantize_name)
+                self._op_names.append(fake_quantize_layer.op_name)
+
+                is_custom, layer_info = converter.get_layer_info_for_node(target_node_name)
+                if is_custom:
+                    raise RuntimeError("Quantizing custom layer activations is currently unsupported!")
+                if input_port_id is not None:
+                    target_point = TFBeforeLayer(layer_info.layer_name,
+                                                 instance_idx=layer_info.instance_idx,
+                                                 input_port_id=input_port_id)
+                else:
+                    target_point = TFAfterLayer(layer_info.layer_name,
+                                                instance_idx=layer_info.instance_idx,
+                                                output_port_id=0)
+                qpoint = TFQuantizationPoint(fake_quantize_name, quantizer_spec, target_point)
+                setup.add_quantization_point(qpoint)
+
+        return setup
+
+    def _get_quantizer_propagation_solution(self, nncf_graph: NNCFGraph,
+                                            quantizable_weighted_layer_nodes: List[QuantizableWeightedLayerNode],
+                                            custom_layer_node_names: List[NNCFNodeName]) \
+            -> SingleConfigQuantizerSetup:
+        ip_graph = InsertionPointGraph(nncf_graph,
+                                       [qn.node.node_name for qn in quantizable_weighted_layer_nodes])
+
+        pattern = TF_HW_FUSED_PATTERNS.get_full_pattern_graph()
+        ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern)
+
+        input_preprocessing_nodes = self._get_input_preprocessing_nodes(nncf_graph)
+        input_preprocessing_node_names = [n.node_name for n in input_preprocessing_nodes]
+        ignored_scopes_for_solver = self.ignored_scopes_per_group[QuantizerGroup.ACTIVATIONS] + \
+                                    input_preprocessing_node_names + custom_layer_node_names
```

Comment:
I believe we should inform the user that the custom layers in the `custom_layer_node_names` will be ignored during quantization.

Reply:
Done